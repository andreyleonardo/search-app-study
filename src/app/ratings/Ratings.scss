@import "src/stylesheets/css/_colors";

.ratings {
  display: flex;
  align-items: center;

  &__icon, &__children {
    color: $primary;
    margin: 3px;
  }
}
