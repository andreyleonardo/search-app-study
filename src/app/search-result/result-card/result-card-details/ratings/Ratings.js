import React from 'react';
import { inRange } from 'lodash';
import PropTypes from 'prop-types';
import HappyIcon from './icons/HappyIcon';
import SmileIcon from './icons/SmileIcon';
import NeutralIcon from './icons/NeutralIcon';
import WonderingIcon from './icons/WonderingIcon';
import SadIcon from './icons/SadIcon';
import colors from '../../../../../stylesheets/css/_colors.scss';

import './Ratings.scss';

const Ratings = ({ rating, children }) => (
  <div className="ratings">
    <span className="ratings__icon">
      {inRange(rating, 0, 1.1) && <SadIcon fill={colors.red} />}
      {inRange(rating, 1.1, 2.1) && <WonderingIcon fill={colors.orange} />}
      {inRange(rating, 2.1, 3.1) && <NeutralIcon fill={colors.yellow} />}
      {inRange(rating, 3.1, 4.1) && <SmileIcon fill={colors.blue} />}
      {inRange(rating, 4.1, 5.1) && <HappyIcon fill={colors.green} />}
    </span>
    {children && (
      <div className="ratings__children">
        { children }
      </div>
    )}
  </div>
);

Ratings.propTypes = {
  rating: PropTypes.number.isRequired,
  children: PropTypes.node
};

Ratings.defaultProps = {
  children: null
};

export default Ratings;
