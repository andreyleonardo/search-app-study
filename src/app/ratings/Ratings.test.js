import React from 'react';
import Ratings from './Ratings';

const getRatings = (props, children) => (
  shallow(<Ratings {...props}>{children}</Ratings>)
);

describe('Ratings', () => {
  it('renders without crash', () => {
    const ratings = getRatings({ rating: 0 });

    expect(ratings).toMatchSnapshot();
  });

  it('renders a SadIcon when rating is between 0 and 1', () => {
    const ratings = getRatings({ rating: 0 });

    expect(ratings.exists('SadIcon')).toBeTruthy();
  });

  it('renders a WonderingIcon when rating is between 1.1 and 2', () => {
    const ratings = getRatings({ rating: 1.1 });

    expect(ratings.exists('WonderingIcon')).toBeTruthy();
  });

  it('renders a NeutralIcon when rating is between 2.1 and 3', () => {
    const ratings = getRatings({ rating: 2.1 });

    expect(ratings.exists('NeutralIcon')).toBeTruthy();
  });

  it('renders a SmileIcon when rating is between 3.1 and 4', () => {
    const ratings = getRatings({ rating: 3.1 });

    expect(ratings.exists('SmileIcon')).toBeTruthy();
  });

  it('renders a HappyIcon when rating is between 4.1 and 5', () => {
    const ratings = getRatings({ rating: 4.1 });

    expect(ratings.exists('HappyIcon')).toBeTruthy();
  });

  it('renders a given children component', () => {
    const ratings = getRatings({ rating: 4.1 }, <span id="children-id" />);

    expect(ratings.exists('span#children-id')).toBeTruthy();
  });
});
