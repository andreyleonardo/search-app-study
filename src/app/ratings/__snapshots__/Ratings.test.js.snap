// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ratings renders without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ratings
    rating={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="ratings__icon"
        >
          <SadIcon
            height="32"
            viewBox="0 0 32 32"
            width="32"
          />
        </span>,
        null,
      ],
      "className": "ratings",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <SadIcon
              height="32"
              viewBox="0 0 32 32"
              width="32"
            />,
            false,
            false,
            false,
            false,
          ],
          "className": "ratings__icon",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "fill": undefined,
              "height": "32",
              "viewBox": "0 0 32 32",
              "width": "32",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          false,
          false,
          false,
        ],
        "type": "span",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="ratings__icon"
          >
            <SadIcon
              height="32"
              viewBox="0 0 32 32"
              width="32"
            />
          </span>,
          null,
        ],
        "className": "ratings",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <SadIcon
                height="32"
                viewBox="0 0 32 32"
                width="32"
              />,
              false,
              false,
              false,
              false,
            ],
            "className": "ratings__icon",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "fill": undefined,
                "height": "32",
                "viewBox": "0 0 32 32",
                "width": "32",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
            false,
            false,
            false,
          ],
          "type": "span",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
